add package.json

VernL opened the pull request:
This pull request will close #31

RichardLitt commented on package.json:
```diff
@@ -0,0 +1,25 @@
+  "keywords": ["independent", "bookstores", "index"],
+  "author": "Richard Littauer",
+  "license": "MIT",
+  "contributors": ["Joanna Katsifaraki", "Jose Esplana", "Hanie Jafari, Vernon Lillies" ],
```
I believe these need to include more information. Take a look at some other package.json's that have it. You can find them through GitHub search. Same for the `author` field.

RichardLitt commented on package.json:
```diff
@@ -0,0 +1,25 @@
+  "bugs": {
+    "url": "https://github.com/RichardLitt/the-travel-shelf/issues"
+  },
+  "homepage": "https://github.com/RichardLitt/the-travel-shelf#readme",
```
No need for `#readme`. 

Learn how to remove this and amend the commit. It's ok to force push to a branch (but _only_ to a branch_).

RichardLitt commented on package.json:
```diff
@@ -0,0 +1,25 @@
+  "version": "1.0.0",
+  "description": "A magazine and index of beautiful global independent bookstores",
+  "main": "index.js",
+  "directories": {
```
Probably not needed. What does this do?

RichardLitt commented:
@VernL Can you amend the commit and the pull request to be lowercase? Read this for how to write great commits: https://gist.github.com/stephenparish/9941e89d80e2bc58a153.

Jojokats commented:
I'm having issues with pushing the changes. I changed the name from app to Base but after doing a git status, it's showing me in red that everything is deleted. 
If you can have a look and let me know what to do. Thanks

![git-m](https://user-images.githubusercontent.com/10158712/31182262-618a59a0-a8f1-11e7-9391-18cd4c18bdcc.png)

RichardLitt commented:
@Jojokats Wrong issue, I think?

Jojokats commented:
>@Jojokats Wrong issue, I think?

I don't understand....

RichardLitt commented:
@Jojokats I think you posted this comment in the wrong PR. It's not relevant to #35.

Jojokats commented:
oh no

VernL commented:
I broke it :D

VernL commented:
@RichardLitt I ran into a merge conflict. I now have two redundant commits. Is there an easy way get rid of them?

RichardLitt commented:
Time to learn about squashing!

How did you get a merge conflict? I'm curious. You shouldn't have switched branches.

RichardLitt commented:
Another option is to reset the branch to the master HEAD while keeping the files in the index, commit on top again, and then force push to this branch with the new commit only.

VernL commented:
I'm curious too :D I was dual wielding command line interface and GitHub Desktop.
I modified  the package.json, ran git commit --amend. Everything going well. Then I switched over to GitHub Desktop to push the changes, but was out of sync. I fetched origin and pulled the changes and all hell broke loose. 

Noted about the branch change.  Should I kill this branch and submit a new pull request?

In the meantime, on to squashing.

VernL commented:
It worked 👍

RichardLitt commented:
> I fetched origin and pulled the changes and all hell broke loose.

You might have fetched origin master, by accident. That would mess things up.

RichardLitt commented on package.json:
```diff
@@ -0,0 +1,24 @@
+
+  "author": "Richard Littauer <richard.littauer@gmail.com> (https://burntfen.com)",
+  "license": "MIT",
+  "contributors": ["Joanna Katsifaraki", "Jose Esplana <jmesplana@gmail.com> (www.jey-em.com)", "Hanie Jafari me@hjafari.com (www.hjafari.com)", "Vernon Lillies <vernon.lillies@gmail.com>" ],
```
Hanie needs `<>` around her email. 

Can you look for a `package.json` linter for me?

RichardLitt commented on package.json:
```diff
@@ -0,0 +1,24 @@
+    "url": "https://github.com/RichardLitt/the-travel-shelf/issues"
+  },
+  "homepage": "https://github.com/RichardLitt/the-travel-shelf",
+
```
Doesn't look valid. It doesn't close. Where is the private tag?

VernL commented:
> Can you look for a package.json linter for me?
online option : http://package-json-validator.com/ 
npm : https://www.npmjs.com/package/package-json-validator

VernL commented:
Should be fixed.

RichardLitt commented on package.json:
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "the-travel-shelf",
+  "version": "1.0.0",
+  "description": "> A magazine and index of beautiful global independent bookstores",
```
No `> `.

VernL replied:
check

RichardLitt commented:
Well done!
